File: VB/WinSample.Module/MyReportParametersObject.vb
Imports Microsoft.VisualBasic
Imports System

Imports DevExpress.Xpo
Imports DevExpress.Xpo.DB
Imports DevExpress.Data.Filtering

Imports DevExpress.ExpressApp
Imports DevExpress.Persistent.Base
Imports DevExpress.Persistent.BaseImpl
Imports DevExpress.Persistent.Validation
Imports DevExpress.ExpressApp.Reports

Namespace WinSample.Module
	<NonPersistent> _
	Public Class MyReportParametersObject
		Inherits ReportParametersObjectBase
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub
		Public Overrides Function GetCriteria() As CriteriaOperator
			Return Nothing
		End Function
		Public Overrides Function GetSorting() As SortingCollection
			Dim sorting As New SortingCollection()
			Return sorting
		End Function
		Private _GroupDepth As Integer
		Public Property GroupDepth() As Integer
			Get
				Return _GroupDepth
			End Get
			Set(ByVal value As Integer)
				SetPropertyValue("GroupDepth", _GroupDepth, value)
			End Set
		End Property
	End Class

End Namespace

File: CS/WinSample.Module/MyReportParametersObject.cs
using System;

using DevExpress.Xpo;
using DevExpress.Xpo.DB;
using DevExpress.Data.Filtering;

using DevExpress.ExpressApp;
using DevExpress.Persistent.Base;
using DevExpress.Persistent.BaseImpl;
using DevExpress.Persistent.Validation;
using DevExpress.ExpressApp.Reports;

namespace WinSample.Module {
    [NonPersistent]
    public class MyReportParametersObject : ReportParametersObjectBase {
        public MyReportParametersObject(Session session) : base(session) { }
        public override CriteriaOperator GetCriteria() {
            return null;
        }
        public override SortingCollection GetSorting() {
            SortingCollection sorting = new SortingCollection();
            return sorting;
        }
        private int _GroupDepth;
        public int GroupDepth {
            get { return _GroupDepth; }
            set { SetPropertyValue("GroupDepth", ref _GroupDepth, value); }
        }
    }

}